Code/GraphMol/MolStandardize/Validate.h

```diff
@@ -191,6 +194,107 @@ class RDKIT_MOLSTANDARDIZE_EXPORT DisallowedAtomsValidation
+  }
+};
+
+//! The FeaturesValidation class reports an error if the input
+/// molecule representation includes any undesired features.
+class RDKIT_MOLSTANDARDIZE_EXPORT FeaturesValidation : public ValidationMethod {
+ public:
+  FeaturesValidation(bool allowEnhancedStereo = false,
+                     bool allowAromaticBondType = false,
+                     bool allowDativeBondType = false,
+                     bool allowQueries = false, bool allowDummies = false,
+                     bool allowAtomAliases = false)
+      : allowEnhancedStereo(allowEnhancedStereo),
+        allowAromaticBondType(allowAromaticBondType),
+        allowDativeBondType(allowDativeBondType),
+        allowQueries(allowQueries),
+        allowDummies(allowDummies),
+        allowAtomAliases(allowAtomAliases){};
+  std::vector<ValidationErrorInfo> validate(
+      const ROMol &mol, bool reportAllFailures) const override;
+  std::shared_ptr<ValidationMethod> copy() const override {
+    return std::make_shared<FeaturesValidation>(*this);
+  }
+  bool allowEnhancedStereo;
+  bool allowAromaticBondType;
+  bool allowDativeBondType;
+  bool allowQueries;
+  bool allowDummies;
+  bool allowAtomAliases;
+};
+
+//! The Is2DValidation class reports an error if the input
+/// molecule representation is designated as 3D or if it includes
+/// non-null Z coordinates, and in case all atoms are assigned the
+/// same coordinates.
+class RDKIT_MOLSTANDARDIZE_EXPORT Is2DValidation : public ValidationMethod {
+ public:
+  Is2DValidation(double threshold = 1.e-3) : threshold(threshold){};
+  std::vector<ValidationErrorInfo> validate(
+      const ROMol &mol, bool reportAllFailures) const override;
+  std::shared_ptr<ValidationMethod> copy() const override {
+    return std::make_shared<Is2DValidation>(*this);
+  }
+
+  double threshold;
+};
+
+//! The Layout2DValidation class reports an error if any atoms are
+/// too close to any other atoms or bonds, and in case any bonds are
+/// too long.
+class RDKIT_MOLSTANDARDIZE_EXPORT Layout2DValidation : public ValidationMethod {
+ public:
+  Layout2DValidation(double clashLimit = 0.15, double bondLengthLimit = 25.,
+                     bool allowLongBondsInRings = true,
+                     bool allowAtomBondClashExemption = true,
+                     double minMedianBondLength = 1e-3)
+      : clashLimit(clashLimit),
+        bondLengthLimit(bondLengthLimit),
+        allowLongBondsInRings(allowLongBondsInRings),
+        allowAtomBondClashExemption(allowAtomBondClashExemption),
+        minMedianBondLength(minMedianBondLength){};
+  std::vector<ValidationErrorInfo> validate(
+      const ROMol &mol, bool reportAllFailures) const override;
+  std::shared_ptr<ValidationMethod> copy() const override {
+    return std::make_shared<Layout2DValidation>(*this);
+  }
+
+  static double squaredMedianBondLength(const ROMol &mol,
+                                        const Conformer &conf);
```

Comment:
I can't simply change it into a free function in an anonymous namespace, because it's at the moment also called by the Pipeline's cleanup 2D operation (in order to determine the scale factor).

In general, I admit I didn't put very much thinking into how to design this function, because I also had the impression these utility functions might belong to some other place altogether. I think the depiction code implement some similar functionality, but (iirc) it wasn't straightforward to reuse.